src/animated/Controller.ts

```diff
@@ -450,14 +439,18 @@ class Controller<DS extends object = any> {
     )
 
     if (changed) {
-      this._willStart(props, oldProps)
+      if (started.length && onStart) {
+        started.forEach(name => onStart!(this.animations[name]))
```

Comment:
Call `onStart` once per animated key, whenever a key has a new goal value and is not immediate. @drcmda Does this sound good?

This definitely messes with `useTransition`, so I've gotta figure out a workaround there.

Reply:
Perhaps this would be better as a new `onAnimation` prop.

Then I'll keep `onStart` the same, where it's only called when the first animation of an "update" is about to start.

Reply:
Actually, I think the utility of `onStart`'s current behavior is dubious. You can always throttle `onStart` if you only want it called once per update, which is what `useTransition` will do.

Reply:
I never like onStart tbh, these prop-triggered lifecycles are all vague

Reply:
Beyond debugging, I don't think `onStart` has much utility. That's why I changed it to be called for every `Animation` object that is about to start.

Reply:
i agree, docs are a sore spot still